### PR TITLE
Match request  cancel upon socket disconnect

### PR DESCRIPTION
**Pull request by javinchua**

### CODE REVIEW

File: project/apps/matching-service/src/db/match.redis.ts [hunk 1]
```diff
@@ -283,4 +291,42 @@ export class MatchRedis {
...
+      this.logger.error(`Error setting userId to matchId: ${error}`);
+    }
+  }
+
+  /**
+   * Removes a user match mapping
+   * @param userId The userId to map
+   */
+
+  async removeUserMatchMapaping(userId: string) {
```
Comment (firwer) [inline project/apps/matching-service/src/db/match.redis.ts:315]: typo haha

**Review by firwer — APPROVED** [review]
lgtm

File: project/apps/matching-service/src/db/match.redis.ts [hunk 2]
```diff
@@ -90,7 +94,7 @@ export class MatchRedis {
 
   /**
    * Fetches a match request from redis
-   * @param match_req_id The matchReqId to fetch
+   * @param match_req_id The match_req_id to fetch
    * @returns The match request if found, otherwise null
    */
   async getMatchRequest(match_req_Id: string): Promise<MatchRequestDto | null> {
```
Comment (jeffrey-jian) [inline project/apps/matching-service/src/db/match.redis.ts:100]: unrelated:
could you also help to remove this log in this PR

https://github.com/CS3219-AY2425S1/cs3219-ay2425s1-project-g30/blob/7d2c8cf13706d41d1c21dd852398962c5baa8b02/project/apps/matching-service/src/db/match.redis.ts#L105

File: project/apps/matching-service/src/db/match.redis.ts [hunk 3]
```diff
@@ -283,4 +291,42 @@ export class MatchRedis {
...
+
+  /**
+   * Removes a user match mapping
+   * @param userId The userId to map
+   */
+
+  async removeUserMatchMapaping(userId: string) {
+    try {
+      const key = `${MATCH_USER}-${userId}`;
+      await this.redisClient.del(key);
```
Comment (jeffrey-jian) [inline project/apps/matching-service/src/db/match.redis.ts:318]: nit: should aim to log any db write operations like delete
`this.logger.log(`Removed userMatchMapping for user ...`

File: project/apps/matching-service/src/matching.gateway.ts [hunk 4]
```diff
@@ -37,6 +38,7 @@ export class MatchingGateway
   constructor(
```
Comment (firwer) [inline project/apps/matching-service/src/matching.gateway.ts:38]: unrelated to current changes: Would it be good to just add in something like
```
  afterInit(server: Server) {
    this.logger.log('WebSocket gateway initialized');
  }
```
for the purpose of sanity check?

**Pull request merged**